src/main/java/org/gitlab4j/api/GitLabApiClient.java

```diff
@@ -273,29 +274,44 @@ void setSudoAsId(Integer sudoAsId) {
 
     /**
      * Construct a REST URL with the specified path arguments.
-     * 
+     *
      * @param pathArgs variable list of arguments used to build the URI
      * @return a REST URL with the specified path arguments
      * @throws IOException if an error occurs while constructing the URL
      */
     protected URL getApiUrl(Object... pathArgs) throws IOException {
+        String url = appendPathArgs(this.hostUrl, pathArgs);
+        return (new URL(url));
+    }
+
+    /**
+     * Construct a REST URL with the specified path arguments using
+     * Gitlab base url.
+     *
+     * @param pathArgs variable list of arguments used to build the URI
+     * @return a REST URL with the specified path arguments
+     * @throws IOException if an error occurs while constructing the URL
+     */
+    protected URL getUrlWithBase(Object... pathArgs) throws IOException {
+        String url = appendPathArgs(this.baseUrl, pathArgs);
+        return (new URL(url));
+    }
 
-        StringBuilder url = new StringBuilder();
-        url.append(hostUrl);
+    private String appendPathArgs(String url, Object... pathArgs) {
+        StringBuilder urlBuilder = new StringBuilder(url);
         for (Object pathArg : pathArgs) {
             if (pathArg != null) {
-                url.append("/");
-                url.append(pathArg.toString());
+                urlBuilder.append("/");
+                urlBuilder.append(pathArg.toString());
             }
         }
-
-        return (new URL(url.toString()));
+        return urlBuilder.toString();
```

Comment:
factored out url building logic so that it can be applied with the baseUrl as well.